modules/core/src/matmul.cpp

```diff
@@ -864,73 +864,39 @@ static bool ocl_gemm( InputArray matA, InputArray matB, double alpha,
     return k.run(2, globalsize, block_size!=1 ? localsize : NULL, false);
 }
 #endif
-}
 
-void cv::gemm( InputArray matA, InputArray matB, double alpha,
-           InputArray matC, double beta, OutputArray _matD, int flags )
+static void _gemmImplInternal( Mat A, Mat B, double alpha,
```

Comment:
It is better to avoid names starting from `_`. I'd suggest renaming this function to `gemmImpl` and `_callInternalGemmImpl` to `callGemmImpl`.